x-pack/plugins/apm/ftr_e2e/cypress/integration/read_only_user/errors/errors_page.spec.ts

```diff
@@ -110,3 +110,66 @@ describe('Errors page', () => {
     });
   });
 });
+
+describe('Check detailed statistics API with multiple errors', () => {
+  before(async () => {
+    cy.loginAsViewerUser();
+    await synthtrace.index(
+      generateMultipleErrorsData({
+        from: new Date(start).getTime(),
+        to: new Date(end).getTime(),
+      })
+    );
+  });
+
+  after(async () => {
+    await synthtrace.clean();
+  });
+
+  it('calls detailed API with visible items only', () => {
+    cy.intercept(
+      'GET',
+      '/internal/apm/services/opbeans-java/errors/groups/main_statistics?*'
+    ).as('errorsMainStatistics');
+    cy.intercept(
+      'POST',
+      '/internal/apm/services/opbeans-java/errors/groups/detailed_statistics?*'
+    ).as('errorsDetailedStatistics');
+    cy.visit(`${javaServiceErrorsPageHref}&pageSize=10`);
+    cy.wait('@errorsMainStatistics');
+    cy.get('.euiPagination__list').children().should('have.length', 5);
+    cy.wait('@errorsDetailedStatistics').then((payload) => {
+      expect(payload.request.body.groupIds).eql(
+        JSON.stringify([
+          '0000000000000000000000000Error 0',
+          '0000000000000000000000000Error 1',
+          '0000000000000000000000000Error 2',
+          '0000000000000000000000000Error 3',
+          '0000000000000000000000000Error 4',
+          '0000000000000000000000000Error 5',
+          '0000000000000000000000000Error 6',
+          '0000000000000000000000000Error 7',
+          '0000000000000000000000000Error 8',
+          '0000000000000000000000000Error 9',
+        ])
+      );
+    });
```

Comment:
This might be technically correct but it's not super clear what this verifies. The test says "calls detailed API with visible items only". 

There are 5 visible items afaict on line 140. Yet, detailed statistics requests 10 values. Isn't that a disconnect? Should those two be the same?

Reply:
No, there are 5 pages, with 10 items on each of them.

Reply:
Okay I missed that. But that's also my point: it's not clear what this is verifying (at least not to me). Why are we asserting the number of pages in the pagination? Is that important when ensuring that it only makes requests for the visible items? Perhaps it's useful to assert that there are more pages than the current one.
Most importantly, I think you should make the connection between `pageSize` and `groupIds` more obvious:


```
    const pageSize = 10;
    cy.visit(`${javaServiceErrorsPageHref}&pageSize=${pageSize}`);
    cy.wait('@errorsMainStatistics');
    cy.get('.euiPagination__list').children().should('be.gt', 1);
    cy.wait('@errorsDetailedStatistics').then((payload) => {
      expect(payload.request.body.groupIds.length).eql(pageSize)
      );
    });
```